components/brave_wallet/browser/eth_response_parser.cc

```diff
@@ -12,9 +12,21 @@
 #include "brave/components/brave_wallet/browser/json_rpc_response_parser.h"
 #include "brave/components/brave_wallet/common/eth_address.h"
 #include "brave/components/brave_wallet/common/hex_utils.h"
+#include "net/base/data_url.h"
 
 namespace brave_wallet {
 
+bool ParseStringResult(const std::string& json, std::string* value) {
```

Comment:
could you add some tests for this? including failure cases like when `result` length is too short

Reply:
Added some in https://github.com/brave/brave-core/pull/12594/commits/1fee326b89606457de71e74ce826279783bacece.

cc @yrliou since I moved from anonymous namespace to make visible from tests.